misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormatterUtils.java

```diff
@@ -15,25 +15,129 @@
 
+    public static Map<String, Object> getFormattingConfigurations(Path root, String path) throws FormatterException {
+        String content;
+        if (isLocalFile(path)) {
+            try {
+                content = Files.readString(Path.of(path), StandardCharsets.UTF_8);
+            } catch (IOException e) {
+                throw new FormatterException("Failed to retrieve local formatting configuration file");
+            }
+        } else {
+            content = readRemoteFormatFile(root, path);
+        }
+
+        TomlDocument document = TomlDocument.from(path, content);
+        TomlTableNode tomlAstNode = document.toml().rootNode();
+
+        Map<String, Object> formatConfigs = new HashMap<>();
+        if (!tomlAstNode.entries().isEmpty()) {
+            Map<String, Object> tomlMap = document.toml().toMap();
+            for (Map.Entry<String, Object> entry : tomlMap.entrySet()) {
+                formatConfigs.put(entry.getKey(), entry.getValue());
+            }
+        }
+        return formatConfigs;
+    }
+
+    public static boolean isLocalFile(String path) {
+        return new File(path).exists();
+    }
+
+    static String readRemoteFormatFile(Path root, String fileUrl) throws FormatterException {
+        StringBuilder fileContent = new StringBuilder();
+        Path cachePath = root.resolve("target").resolve("format").resolve("Format.toml");
+
+        if (Files.exists(cachePath)) {
+            try {
+                return Files.readString(cachePath, StandardCharsets.UTF_8);
+            } catch (IOException e) {
+                throw new FormatterException("Failed to read cached formatting configuration file");
+            }
+        }
+
+        try {
+            URL url = new URL(fileUrl);
+            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+
+            int responseCode = connection.getResponseCode();
+
+            if (responseCode == HttpURLConnection.HTTP_OK) {
+                try (BufferedReader reader = new BufferedReader(
+                        new InputStreamReader(connection.getInputStream(), StandardCharsets.UTF_8))) {
+                    String line;
+                    while ((line = reader.readLine()) != null) {
+                        fileContent.append(line).append(NEWLINE_SYMBOL);
+                    }
+                }
+            } else {
+                throw new FormatterException("Failed to retrieve remote file. HTTP response code: " + responseCode);
+            }
+            connection.disconnect();
+        } catch (IOException e) {
+            throw new FormatterException("Failed to retrieve remote formatting configuration file");
+        }
+        cacheRemoteConfigurationFile(root, fileContent.toString());
+
+        return fileContent.toString();
+    }
+
+    private static void cacheRemoteConfigurationFile(Path root, String content) throws FormatterException {
+        if (Files.exists(root.resolve("target"))) {
```

Comment:
Better if we can return when target not found.

```suggestion
        if (!Files.exists(root.resolve("target"))) {
          return;
        }
```

Reply:
If the target is not found it doesnt go instead the if condition and and returns without us having having to explicitly code it. Existiing method avoids having `return;` code line